controllers/tinkerbell_datacenter_controller.go

```diff
@@ -0,0 +1,20 @@
+package controllers
+
+import (
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TinkerbellDatacenterReconciler reconciles a TinkerbellDatacenterConfig object.
+type TinkerbellDatacenterReconciler struct {
+	log    logr.Logger
```

Comment:
better to extract the logger from the context in the way the cluster controller does it
That ensures we inherit the values set controller-runtime